tests/pytest_framework/src/syslog_ng/syslog_ng.py

```diff
@@ -35,8 +35,8 @@ def __init__(self, logger_factory, instance_paths, syslog_ng_ctl):
     def start(self, config, external_tool=None):
         self.__syslog_ng_cli.start(config, external_tool)
 
-    def stop(self, unexpected_messages=None):
-        self.__syslog_ng_cli.stop(unexpected_messages)
+    def stop(self, unexpected_messages=None, signal=None):
```

Comment:
This patch is not clear for me. I have a few questions for clarifications, because I do not see the use case here.
- Why do we want to stop syslog-ng with signal.
- How can the users utilize this? Can you add an example testcase with assertions.
- Do we want to allow any kind of signals, like sigchild?
- Can we avoid extending the `stop` method with this parameter?

It would worth put this patch into a separate PR.

Reply:
I am also not really sure to support this case.
Example testcase:
import signal
config = some_dummy_config()
syslog_ng = tc.new_syslog_ng()
syslog_ng.start(config)
syslog_ng.stop(signal=signal.SIGSEGV)

With such steps we can test if syslog-ng saves internal states/persist correctly. 

Is it ok?

Reply:
For that case, I think creating a `syslog_ng.send_signal(signal.SIGSEGV)` would be more clear. Syslog-ng does not necessarily stop for any kind of signals, so linking signal management with stop is not a good coupling.

Reply:
+1, will be fixed for this way